Kudu.Client/Infrastructure/HttpClientExtensions.cs

```diff
@@ -93,5 +93,22 @@ public static T GetJson<T>(this HttpClient client, string url)
 
             return JsonConvert.DeserializeObject<TOutput>(content);
         }
+
+        public static async Task<HttpResponseResult<TOutput>> RichPutJsonAsync<TInput, TOutput>(this HttpClient client, string url, TInput param)
```

Comment:
I would just do it in the existing PutJsonAsync.  If we detect that TOutput type is HttpResponseResult (or certain interface), we always add response headers.   It is transparent to callers.